jupyter_core/paths.py

```diff
@@ -20,6 +20,7 @@
 import functools
 import traceback
 import importlib
+import importlib_metadata
```

Comment:
You can make this use stdlib on newer pythons as in  https://github.com/pypa/virtualenv/blob/adcf327ee507498a613dc6f235c24c8f2576b6c6/src/virtualenv/run/plugin/base.py#L6-L9